pkg/reconciler/revision/background.go

```diff
@@ -157,54 +156,70 @@ func (r *backgroundResolver) Resolve(logger *zap.SugaredLogger, rev *v1.Revision
 	if !inFlight {
 		logger.Debugf("Adding Resolve request to queue (depth: %d)", r.queue.Len())
 		r.addWorkItems(rev, name, opt, registriesToSkip, timeout)
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	if !result.ready() {
 		logger.Debug("Resolve request in flight, returning nil, nil")
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	ret := r.results[name]
 	if ret.err != nil {
 		logger.Debugf("Resolve returned the resolved error: %v", ret.err)
-		return nil, ret.err
+		return nil, nil, ret.err
 	}
 
-	array := make([]v1.ContainerStatus, len(rev.Spec.Containers))
-	for k, v := range ret.statuses {
-		array[k] = v
+	initContainerStatuses = make([]v1.ContainerStatus, len(rev.Spec.InitContainers))
+	for i, container := range rev.Spec.InitContainers {
+		if resolvedDigest, ok := ret.imagesResolved[container.Image]; ok {
```

Comment:
is it actually possible for imagesResolved not to have an entry here? If not I might drop the if/ok